### PR TITLE
Exercise bikes!

### PR DESCRIPTION
**Pull request by RemieRichards**
<B>Details:</B>
- Adds exercise bikes, a new object that produces power (1/4 of a basic portagen at the moment, 1250) 
- Riding on an exercise bike lowers your nutrition (exercise! it causes weight loss!) which is both fluff and to help you realise you're actually pedalling.
- Can be wrenched/unwrenched so they can be moved, need to be wrenched down to pedal+power
- Can be ordered from cargo for 25 points

<B>Mappers:</B>
I suggest they be placed in perma, so that the prisoners can do something useful for the station, pedal!
Although feel free to map them wherever you please (Public Gym anyone?)

<B>How to ride like a pro:</B>
Alternate between the EAST/WEST movement keys (A/D, Arrow keys etc.)
You can tell which one to press based on which pedal is up (Coming soon with sprites)

<B>TODO:</B>
- [ ] Sprite
- [ ] Adjust update_visuals() to work with new sprite
- [ ] Replace debug messages with a visual update

### CODE REVIEW

**Comment by paprka** [comment]
Should be more fun like you pedal back and forth, A + D or < + > to generate power

**Comment by hornygranny** [comment]
powergame exercise bike click macro

**Comment by paprka** [comment]
I'm adding a weight bench i swear

**Comment by RemieRichards** [comment]
@paprka I'm really tempted to add that now... but how would you inform the player they're pressing the right button at the right time?

File: code/modules/power/exercise_bikes/exerciseBike.dm [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+	if(direction == dir)
+		producing = 1
+		user.nutrition = max(user.nutrition - 2, 0)
+
+	update_visuals()
+
+
+/obj/machinery/power/exercise_bike/process()
+	if(producing)
+		add_avail(1250)
```
Comment (Iamgoofball) [inline code/modules/power/exercise_bikes/exerciseBike.dm]: make this into a var

**Comment by Boredone** [comment]
Out of curiosity, what would happen if you buckled in a random NPC monkey?

**Comment by RemieRichards** [comment]
@Boredone it'd randomly walk around, so with the old method of power gen (pressing the dir of the bike) it'd have a 1/4 chance of actually powering it once, with the new method it'd have a 1/8 chance, but yeah I guess you monkeys could use it, Haha.

File: code/modules/power/exercise_bikes/exerciseBike.dm (hunk 1 above)
Comment (RemieRichards) [inline code/modules/power/exercise_bikes/exerciseBike.dm]: Upgradable exercise bikes? heh, sure ok.

**Comment by hornygranny** [comment]
monkey exercise bike farm confirmed

**Comment by GunHog** [comment]
We should add this to walking as well! Full on SS13 QWOP! Honestly we should be able to build these, or get more from cargo. Honk.

**Comment by ike709** [comment]
>REMIERICHARDS STILL HAS NOT FINISHED COMPUTERS OR DISMEMBERMENT

:+1: x :100: now I can remove my blubber!

**Comment by RemieRichards** [comment]
@GunHog you may now:
* Order from cargo for 25 points
* Wrench/Unwrench the bike to move it around

**Comment by ike709** [comment]
About how many would it take to power the station?

**Comment by tkdrg** [comment]
About 250, I'd say. That'd give you over 300kW.

**Comment by Tenebrosity** [comment]
Did Black Mirror make you think of these?

These are rarely going to get used, by the way.

**Comment by Alek2ander** [comment]
Next step: randomized RPG stats so you have "endurance" that gets upgraded when you pedal. Almost like a **real MMO**!

**Comment by RemieRichards** [comment]
@Tenebrosity not really, I saw that episode of black mirror ages ago, literal ages.
and I don't really care about their usage, they're just something fun than can be used for RP, Fat burning, Power gen or a mix of the 3.

**Comment by Lumbermancer** [comment]
>exercise! it causes weight loss!

What's the point? I haven't see a fat person on this server literally in years (remember when fatties would explode losing clothes?). Can you even get fat just by eating anymore?

**Comment by drynwyn** [comment]
You can get fat.

There just aren't fatsprites.

**Comment by RemieRichards** [comment]
Fatness was never removed, it slows you down and changes your examine text.
it was fatsprites that were removed.

**Comment by Miauw62** [comment]
/fit/station 13

**Comment by Lumbermancer** [comment]
>Fatness was never removed, it slows you down and changes your examine text.
it was fatsprites that were removed.

How horrible. Add fatsprites along bikes.

**Comment by Kromgar** [comment]
What is the fucking point of this? This will likely never be used

**Comment by hornygranny** [comment]
if you don't know what this could be used for you're not robust

**Comment by Dorsisdwarf** [comment]
Should be able to be hacked to launch off their castors and knock down anyone in a straight line, so prisoners can use them to roflstomp sec.

**Comment by ike709** [comment]
should be able to emag them to somehow allow you to ride them around the station using spessman logic

**Comment by AlexCrimson** [comment]
So basically, command can just enslave the Assistants and use them to power the station? If so, i am totally ok with this.

**Comment by RemieRichards** [comment]
@AlexCrimson anything with the intelligence to press EAST then WEST then EAST etc.
so sadly not Assistants, Hehe.

**Comment by ike709** [comment]
Maybe we can use the SNPCs as part of an admin event to power to station with these bikes.

**Comment by Jalleo** [comment]
Monkey power event sounds like fun.

**Comment by Tenebrosity** [comment]
It sounds like you did get the idea from Black Mirror then.


To be honest, these are just going to be 5 minute curiosities that people stop using the minute they realise they don't really serve a purpose.

**Comment by ike709** [comment]
@Tenebrosity Like new away missions?

**Comment by Tenebrosity** [comment]
I don't mean to seem against this, or derail the PR comments, it just seems like something that won't really get used.

>Like new away missions?

Yeah, exactly.

**Comment by ZergRush000** [comment]
@ike709 
-Emag them to ride around the station at sanic speeds
-Running over people causes some brute damage (like 2.5 to prevent massive amounts of griff)
perfect

**Comment by Tenebrosity** [comment]
That would be pretty amazing. Not to mention (fun).

—

File: code/modules/power/exercise_bikes/exerciseBike.dm [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+	if(direction == required_dir)
+		user.nutrition = max(user.nutrition - 2, 0)
+		add_avail(produced_power)
+		switch(required_dir)
+			if(EAST)
+				required_dir = WEST
+				world << "PRESS WEST"
+			else
+				required_dir = EAST
+				world << "PRESS EAST"
```
Comment (KazeEspada) [inline code/modules/power/exercise_bikes/exerciseBike.dm]: world? Now the entire world knows your fat.

Comment (RemieRichards) [inline code/modules/power/exercise_bikes/exerciseBike.dm]: Lovely to see you back @KazeEspada 
From my checklist on the PR:
>Replace debug messages with a visual update

**Comment by ike709** [comment]
iiiiiii neeeeeddd itttttt

**Comment by Jordie0608** [comment]
Changelog.

**Comment by Niknakflak** [comment]
Im very looking forward to this being added.

**Comment by Niknakflak** [comment]
Still hoping this will get finished one day

**Comment by PKPenguin321** [comment]
RIP

**Comment by ike709** [comment]
sometimes I like to look at the really old PRs and wonder if they'll ever be finished & merged

**Comment by Razharas** [comment]
Ok RR is a maintainer if they decide to finish it they can reopen it

**Pull request closed without merging**